### PR TITLE
fix secret expected length to be at least (1)

### PR DESCRIPTION
**Pull request by cchien816**
Signed-off-by: Chris Chien <chris.chien@suse.com>

From https://github.com/longhorn/longhorn/issues/3892
Newest terraform not accept length = 0 cluster secret, modify code to cope it

cc @meldafrawi

### CODE REVIEW

File: test_framework/terraform/aws/sles/main.tf [hunk 1]
```diff
@@ -250,3 +250,8 @@ resource "aws_eip" "lh_aws_eip_controlplane" {
   count    = var.lh_aws_instance_count_controlplane
   vpc      = true
 }
+
+resource "random_password" "cluster_secret" {
+  length = 64
+  special = false
+}
```
Comment (innobead) [inline test_framework/terraform/aws/sles/main.tf:257]: Could you move this under the below resource call, and add a comment (ex: `Create a random string for the cluster secret`) for it. 

```hcl
resource "random_string" "random_suffix" {
...
}

<-- add here
```

**Review by innobead — CHANGES_REQUESTED** [review]
Good job!

Please also apply this change to other distros as well.

File: test_framework/terraform/aws/sles/main.tf (hunk 1 above)
Comment (cchien816) [reply test_framework/terraform/aws/sles/main.tf]: Hi @innobead ,

I moved the code and added comment. And apply those changes on all distro for review, thank you.

**Review by innobead — APPROVED** [review]

**Comment by innobead** [comment]
@cchien816 See if need to backport to 1.2.x and 1.1.x. I believe we have the same code bases for the testing infra provision.

**Comment by cchien816** [comment]
Hi @innobead ,

I looked into v1.2.x and find that v1.2.x currently only support k3s, hence the password were fixed in length 64. 
We won't hit length = 0 error unless we support rke2 cluster on v1.2.x, thank you.

**Pull request merged**